nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,513 @@
   meta.doc = ./netbird.md;
 
+  imports = [
+    (mkRemovedOptionModule [ "services" "netbird" "tunnels" ]
+      "The option `services.netbird.tunnels` has been renamed to `services.netbird.clients`")
+  ];
+
   options.services.netbird = {
-    enable = mkEnableOption "Netbird daemon";
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Enables backwards compatible Netbird client service.
+
+        This is strictly equivalent to:
+
+        ```nix
+        services.netbird.clients.wt0 = {
+          port = 51820;
+          name = "netbird";
+          interface = "wt0";
+          hardened = false;
+        };
+        ```
+      '';
+    };
     package = mkPackageOption pkgs "netbird" { };
 
-    tunnels = mkOption {
+    ui.enable = mkOption {
+      type = bool;
+      default = config.services.xserver.displayManager.sessionPackages != [ ];
+      defaultText = literalExpression ''config.services.xserver.displayManager.sessionPackages != [ ]'';
+      description = ''
+        Controls presence `netbird-ui` wrappers, defaults to presence of graphical sessions.
+      '';
+    };
+    ui.package = mkPackageOption pkgs "netbird-ui" { };
+
+    clients = mkOption {
       type = attrsOf (
         submodule (
           { name, config, ... }:
           {
             options = {
               port = mkOption {
                 type = port;
-                default = 51820;
+                example = literalExpression "51820";
+                description = ''
+                  Port the Netbird client listens on.
+                '';
+              };
+
+              name = mkOption {
+                type = str;
+                default = "netbird-${name}";
+                description = ''
+                  Primary name for use in:
+                  - systemd service name,
+                  - hardened user name and group,
+                  - [systemd `*Directory=`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectory=) names,
+                  - desktop application identification,
+                '';
+              };
+
+              interface = mkOption {
+                type = str;
+                default = "netbird-${name}";
                 description = ''
-                  Port for the ${name} netbird interface.
+                  Name of the network interface managed by this client.
                 '';
               };
 
               environment = mkOption {
                 type = attrsOf str;
                 defaultText = literalExpression ''
                   {
-                    NB_CONFIG = "/var/lib/''${stateDir}/config.json";
-                    NB_LOG_FILE = "console";
-                    NB_WIREGUARD_PORT = builtins.toString port;
-                    NB_INTERFACE_NAME = name;
-                    NB_DAMEON_ADDR = "/var/run/''${stateDir}"
+                    NB_CONFIG = "/var/lib/''${config.name}/config.json";
```

Comment:
I think freely letting the user choose `name` here is problematic, because it is directly used as a directory name in `/var/lib` or `/run`, which isn't something a user would expect.

If I choose a name similar to another service on the system, their state directories will clash which can cause a lot of issues that we should avoid IMO by always forcing the `netbird-` prefix everywhere $name is used right now, instead of just setting the default to `netbird-${<attrname>}`.

Reply:
Maybe introducing a `stateDir` option that defaults to `netbird-${name}` would suffice.

Reply:
actually forcing a prefix sounds like a good idea